src/com/goide/inspections/GoPlaceholderChecker.java

```diff
@@ -0,0 +1,456 @@
+      this.placeholder = placeholder;
+      this.startPos = startPos;
+      this.position = position;
+      this.verb = verb;
+      this.state = state;
+      this.arguments = arguments;
+      this.flags = flags;
+    }
+
+    public String getPlaceholder() {
+      return placeholder;
+    }
+
+    public int getPosition() {
+      return position;
+    }
+
+    public State getState() {
+      return state;
+    }
+
+    public char getVerb() {
+      return verb;
+    }
+
+    public List<Integer> getArguments() {
+      return arguments;
+    }
+
+    public String getFlags() {
+      return flags;
+    }
+
+    public int getStartPos() {
+      return startPos;
+    }
+  }
+
+  @NotNull
+  public static List<Placeholder> parsePrintf(@NotNull String placeholderText, int firstArg) {
+    List<Placeholder> placeholders = new ArrayList<Placeholder>();
+    int argNum = firstArg;
+    int w;
+    for (int i = 0; i < placeholderText.length(); i += w) {
+      w = 1;
+      if (placeholderText.charAt(i) == '%') {
+        FormatState state = parsePrintfVerb(placeholderText.substring(i), i, firstArg, argNum);
+        placeholders.add(state.toPlaceholder());
+        w = state.format.length();
+
+        // TODO florin: when is this empty??
+        if (!state.argNums.isEmpty()) {
+          if (!state.indexed) {
+            int maxArgNum = Collections.max(state.argNums) + 1;
+            if (argNum < maxArgNum) {
+              argNum = maxArgNum;
+            }
+          }
+          else {
+            argNum = state.argNums.get(state.argNums.size() - 1) + 1;
+          }
+        }
+      }
+    }
+
+    return placeholders;
+  }
+
+  protected static int getPlaceholderPosition(@NotNull GoFunctionOrMethodDeclaration function) {
+    Integer position = FORMATTING_FUNCTIONS.get(StringUtil.toLowerCase(function.getName()));
+    if (position != null) {
+      String importPath = function.getContainingFile().getImportPath(false);
+      if ("fmt".equals(importPath) || "log".equals(importPath) || TESTING_PATH.equals(importPath)) {
+        return position;
+      }
+    }
+    return -1;
+  }
+
+  @Nullable
+  protected static String resolve(@NotNull GoExpression argument) {
+    if (argument instanceof GoStringLiteral) return argument.getText();
+
+    PsiReference reference = argument.getReference();
+    PsiElement resolved = reference != null ? reference.resolve() : null;
+
+    String value = null;
+    if (resolved instanceof GoVarDefinition) {
+      value = getValue(((GoVarDefinition)resolved).getValue());
+    }
+    else if (resolved instanceof GoConstDefinition) {
+      value = getValue(((GoConstDefinition)resolved).getValue());
+    }
+
+    // We always receive the text with double quotes at the beginning and at the end so remove them
+    if (value != null) {
+      value = value.substring(1, value.length() - 1);
```

Comment:
Is there a way to avoid getting the string value double quoted? There'd at least one more place where I had to take this into account to do the check.